shared-module/pathlib/PosixPath.c

```diff
@@ -0,0 +1,333 @@
+        // Return a copy to ensure immutability
+        pathlib_posixpath_obj_t *p = MP_OBJ_TO_PTR(path);
+        return pathlib_posixpath_from_str(p->path_str);
+    }
+
+    const mp_obj_type_t *path_type = mp_obj_get_type(path);
+    mp_raise_TypeError_varg(MP_ERROR_TEXT("%q must be of type %q or %q, not %q"), MP_QSTR_path, MP_QSTR_str, MP_QSTR_PosixPath, path_type->name);
+    return mp_const_none;
+}
+
+mp_obj_t common_hal_pathlib_posixpath_joinpath(pathlib_posixpath_obj_t *self, mp_obj_t arg) {
+
+    mp_obj_t path_part;
+    if (mp_obj_is_str(arg)) {
+        path_part = arg;
+    } else if (mp_obj_is_type(arg, &pathlib_posixpath_type)) {
+        pathlib_posixpath_obj_t *p = MP_OBJ_TO_PTR(arg);
+        path_part = p->path_str;
+    } else {
+        const mp_obj_type_t *arg_type = mp_obj_get_type(arg);
+        mp_raise_TypeError_varg(MP_ERROR_TEXT("%q must be of type %q or %q, not %q"), MP_QSTR_path, MP_QSTR_str, MP_QSTR_PosixPath, arg_type->name);
+        return mp_const_none;
+    }
+
+    // Get string values
+    GET_STR_DATA_LEN(self->path_str, path_str, path_len);
+    GET_STR_DATA_LEN(path_part, part_str, part_len);
+
+    // If either part is empty, return the other
+    if (path_len == 0) {
+        return pathlib_posixpath_from_str(path_part);
+    }
+    if (part_len == 0) {
+        return pathlib_posixpath_from_str(self->path_str);
+    }
+
+    // Calculate the required buffer size (path + '/' + part + '\0')
+    size_t need_slash = (path_str[path_len - 1] != '/') ? 1 : 0;
+    size_t new_len = path_len + need_slash + part_len;
+
+    // Allocate new string
+    byte *new_str = m_new(byte, new_len + 1);
+
+    // Copy first part
+    memcpy(new_str, path_str, path_len);
+
+    // Add slash if needed
+    if (need_slash) {
+        new_str[path_len] = '/';
+    }
+
+    // Copy second part
+    memcpy(new_str + path_len + need_slash, part_str, part_len);
+    new_str[new_len] = '\0';
+
+    // Create a new string object
+    mp_obj_t result_str = mp_obj_new_str((const char *)new_str, new_len);
+
+    // Free the temporary buffer
+    m_del(byte, new_str, new_len + 1);
+
+    return pathlib_posixpath_from_str(result_str);
+}
+
+mp_obj_t common_hal_pathlib_posixpath_parent(pathlib_posixpath_obj_t *self) {
+    GET_STR_DATA_LEN(self->path_str, path_str, path_len);
+
+    // Handle root directory and empty path
+    if (path_len == 0 || (path_len == 1 && path_str[0] == '/')) {
+        return MP_OBJ_FROM_PTR(self); // Root is its own parent
+    }
+
+    // Find the last slash
+    int last_slash = -1;
+    for (int i = path_len - 1; i >= 0; i--) {
+        if (path_str[i] == '/') {
+            last_slash = i;
+            break;
+        }
+    }
+
+    // No slash found, return "."
+    if (last_slash == -1) {
+        return pathlib_posixpath_from_str(MP_OBJ_NEW_QSTR(MP_QSTR_dot));
+    }
+
+    // Root directory
+    if (last_slash == 0) {
+        return pathlib_posixpath_from_str(MP_OBJ_NEW_QSTR(MP_QSTR_slash));
```

Comment:
This line results in the parent being returned as "Slash" which didn't function as a posixpath object. Changing to the following returns "/" which does function properly

```return pathlib_posixpath_from_str(MP_OBJ_NEW_QSTR(qstr_from_str("/")));```